scripts/update.sh

```diff
@@ -2,32 +2,33 @@
 # shellcheck source=/dev/null
 source "/home/steam/server/helper_functions.sh"
 
-UpdateRequired
-updateRequired=$?
 # Check if Update was actually required
-if [ "$updateRequired" != 0 ]; then
-  exit 0
-fi
+if UpdateRequired; then
+    if [ "${UPDATE_ON_BOOT}" = false ]; then
+        LogWarn "An update is available however, UPDATE_ON_BOOT needs to be enabled for auto updating"
+        DiscordMessage "An update is available however, UPDATE_ON_BOOT needs to be enabled for auto updating" "warn"
+        exit 1
+    fi
 
-if [ "${UPDATE_ON_BOOT}" = false ]; then
-    LogWarn "An update is available however, UPDATE_ON_BOOT needs to be enabled for auto updating"
-    DiscordMessage "An update is available however, UPDATE_ON_BOOT needs to be enabled for auto updating" "warn"
-    exit 1
-fi
+    if [ "${RCON_ENABLED,,}" = false ]; then
+        LogWarn "An update is available however auto updating without rcon is not supported"
+        DiscordMessage "An update is available however auto updating without rcon is not supported" "warn"
+        exit 1
+    fi
 
-if [ "${RCON_ENABLED,,}" = false ]; then
-    LogWarn "An update is available however auto updating without rcon is not supported"
-    DiscordMessage "An update is available however auto updating without rcon is not supported" "warn"
-    exit 1
-fi
+    if [[ "${AUTO_UPDATE_WARN_MINUTES}" =~ ^[0-9]+$ ]]; then
+        DiscordMessage "Server will update in ${AUTO_UPDATE_WARN_MINUTES} minutes"
+    fi
```

Comment:
I think we would need to cancel the update if WARN_MINUTES is not an integer.
```suggestion
    if [[ "${AUTO_UPDATE_WARN_MINUTES}" =~ ^[0-9]+$ ]]; then
        LogError "AUTO_UPDATE_WARN_MINUTES is not an integer: ${AUTO_UPDATE_WARN_MINUTES}. Please check your environment variables"
        DiscordMessage "AUTO_UPDATE_WARN_MINUTES is not an integer: ${AUTO_UPDATE_WARN_MINUTES}. Please check your environment variables" "failure"
        exit 1
    fi
    DiscordMessage "Server will update in ${AUTO_UPDATE_WARN_MINUTES} minutes"
```

Reply:
I have it like that because if you so choose you could run the update script with no one in the server and it will update.

Also if it is a cron job and there is no one why not just update? The timer is supposed to just let players know and not prevent the update as that's what the enable flag is for

Reply:
I see what you mean. For me it just seems like we are allowing the user to misconfigure the env variables, which feels wrong. I think we should tell the user to fix it, but maybe we should do that in the start.sh then. To do it immediately and not wait until the update / reboot takes place.

If the user wants to have this that the server restarts immediately he could just set WARN_MINUTES to 0 (not sure if that works as expected, haven't tested that).

Reply:
We should still be checking in backup.sh, restore.sh, and update.sh as those are also intended to be started by the user and not just cron.

I would say for another PR we add validation for all variables which are an expected type such as int, float, boolean.

The implementation in 0.26.0 is no validation for AUTO_UPDATE_WARN_MINUTES.

Reply:
Thinking about it some more auto update and auto reboot should be their own script and then have a separate scripts which are to be ran by the user.

Because in my mind the user should be able to run update and tell it how long to wait or reboot instantly whereas now you can run it but you don't set the values.

This should definitely be a different PR.

Reply:
I was thinking the user interactive script would just be a wrapper for the non interactive script.